Resources/public/js/views/universaldiscovery/ez-universaldiscoverysearchview.js

```diff
@@ -0,0 +1,500 @@
+                };
+            });
+        },
+
+        /**
+         * Refreshes the search result list
+         *
+         * @protected
+         * @method _refresh
+         */
+        _refresh: function () {
+            this._uiPageLoading();
+            this._fireLocationSearch();
+        },
+
+        /**
+         * Sets the UI in the loading the state
+         *
+         * @protected
+         * @method _uiPageLoading
+         */
+        _uiPageLoading: function () {
+            this.get('container').addClass(IS_PAGE_LOADING);
+        },
+
+        /**
+         * Removes the loading state of the UI
+         *
+         * @method _uiPageEndLoading
+         * @protected
+         */
+        _uiPageEndLoading: function () {
+            this.get('container').removeClass(IS_PAGE_LOADING);
+        },
+
+        /**
+         * Fires the `selectContent` event for the given `selection`
+         *
+         * @method _fireSelectContent
+         * @param {Object|Null} selection
+         * @protected
+         */
+        _fireSelectContent: function (selection) {
+            /**
+             * Fired when a content is selected or unselected. The event facade
+             * provides the content structure (the contentInfo, location and content
+             * type models) if a selection was made.
+             *
+             * @event selectContent
+             * @param selection {Object|Null}
+             * @param selection.contentInfo {eZ.ContentInfo}
+             * @param selection.location {eZ.Location}
+             * @param selection.contentType {eZ.ContentType}
+             */
+            this.fire('selectContent', {
+                selection: selection,
+            });
+        },
+
+        /**
+         * `visibleChange` event handler. It makes to reset the current
+         * selection when the search method is hidden/showed
+         *
+         * @method _unselectContent
+         * @protected
+         */
+        _unselectContent: function () {
+            this._fireSelectContent(null);
+            this.get('selectedView').set('contentStruct', null);
+        },
+
+        /**
+         * Preview button `tap` event handler. It prepares contentStruct by taking location from
+         * result list basing on location's id and adding contentInfo and contentType. After that
+         * the row containing selected location is highlighted.
+         *
+         * @method _selectContent
+         * @protected
+         * @param {EventFacade} e
+         */
+        _selectContent: function (e) {
+            var c = this.get('container'),
+                locationId = e.target.getAttribute('data-location-id'),
+                locationStruct = this._getLocationStructFromResultList(locationId),
+                location = locationStruct.location,
+                contentType = locationStruct.contentType,
+                contentInfo = location.get('contentInfo'),
+                contentStruct = {
```

Comment:
here as well, you'll have to add the content when the UDW is configured to load the contents.